diff --git a/logging.h b/logging.h
index 2ddae18..14f4b6a 100644
--- a/logging.h
+++ b/logging.h
@@ -35,11 +35,11 @@ void tgvoip_log_file_write_header(FILE* file);
 
 #include "os/darwin/TGLogWrapper.h"
 
-#define LOGV(msg, ...) {__tgvoip_call_tglog("V/tgvoip: " msg, ##__VA_ARGS__); tgvoip_log_file_printf('V', msg, ##__VA_ARGS__);}
-#define LOGD(msg, ...) {__tgvoip_call_tglog("D/tgvoip: " msg, ##__VA_ARGS__); tgvoip_log_file_printf('D', msg, ##__VA_ARGS__);}
-#define LOGI(msg, ...) {__tgvoip_call_tglog("I/tgvoip: " msg, ##__VA_ARGS__); tgvoip_log_file_printf('I', msg, ##__VA_ARGS__);}
-#define LOGW(msg, ...) {__tgvoip_call_tglog("W/tgvoip: " msg, ##__VA_ARGS__); tgvoip_log_file_printf('W', msg, ##__VA_ARGS__);}
-#define LOGE(msg, ...) {__tgvoip_call_tglog("E/tgvoip: " msg, ##__VA_ARGS__); tgvoip_log_file_printf('E', msg, ##__VA_ARGS__);}
+#define LOGV(msg, ...) {__tgvoip_call_tglog(__FILE__, __FUNCTION__, __LINE__, "V/tgvoip: " msg, ##__VA_ARGS__); tgvoip_log_file_printf('V', msg, ##__VA_ARGS__);}
+#define LOGD(msg, ...) {__tgvoip_call_tglog(__FILE__, __FUNCTION__, __LINE__, "D/tgvoip: " msg, ##__VA_ARGS__); tgvoip_log_file_printf('D', msg, ##__VA_ARGS__);}
+#define LOGI(msg, ...) {__tgvoip_call_tglog(__FILE__, __FUNCTION__, __LINE__, "I/tgvoip: " msg, ##__VA_ARGS__); tgvoip_log_file_printf('I', msg, ##__VA_ARGS__);}
+#define LOGW(msg, ...) {__tgvoip_call_tglog(__FILE__, __FUNCTION__, __LINE__, "W/tgvoip: " msg, ##__VA_ARGS__); tgvoip_log_file_printf('W', msg, ##__VA_ARGS__);}
+#define LOGE(msg, ...) {__tgvoip_call_tglog(__FILE__, __FUNCTION__, __LINE__, "E/tgvoip: " msg, ##__VA_ARGS__); tgvoip_log_file_printf('E', msg, ##__VA_ARGS__);}
 
 #elif defined(_WIN32) && defined(_DEBUG)
 
diff --git a/os/darwin/SetupLogging.h b/os/darwin/SetupLogging.h
index ad10754..960fc9e 100644
--- a/os/darwin/SetupLogging.h
+++ b/os/darwin/SetupLogging.h
@@ -1,3 +1,3 @@
 #import <Foundation/Foundation.h>
 
-extern void (*TGVoipLoggingFunction)(NSString *);
+extern void (*TGVoipLoggingFunction)(const char *, const char *, int, NSString *, va_list args);
diff --git a/os/darwin/TGLogWrapper.h b/os/darwin/TGLogWrapper.h
index 289577e..26d7870 100644
--- a/os/darwin/TGLogWrapper.h
+++ b/os/darwin/TGLogWrapper.h
@@ -11,7 +11,7 @@
 extern "C" {
 #endif
 
-void __tgvoip_call_tglog(const char* format, ...);
+void __tgvoip_call_tglog(const char * filename, const char * functionName, int lineNumber, const char* format, ...);
 
 #if defined __cplusplus
 };
diff --git a/os/darwin/TGLogWrapper.m b/os/darwin/TGLogWrapper.m
index 4739ec8..0e65b59 100644
--- a/os/darwin/TGLogWrapper.m
+++ b/os/darwin/TGLogWrapper.m
@@ -1,13 +1,14 @@
 #import <Foundation/Foundation.h>
 
-void (*TGVoipLoggingFunction)(NSString *) = NULL;
+void (*TGVoipLoggingFunction)(const char *, const char *, int, NSString *, va_list args) = NULL;
 
-void __tgvoip_call_tglog(const char* format, ...){
+void __tgvoip_call_tglog(const char * filename, const char * functionName, int lineNumber, const char* format, ...){
 	va_list args;
 	va_start(args, format);
-    NSString *string = [[NSString alloc] initWithFormat:[[NSString alloc]initWithUTF8String:format] arguments:args];
-	va_end(args);
+
     if (TGVoipLoggingFunction) {
-        TGVoipLoggingFunction(string);
+        TGVoipLoggingFunction(filename, functionName, lineNumber, [[NSString alloc] initWithUTF8String:format], args);
     }
+
+    va_end(args);
 }
